### PR TITLE
Add blake2b/generichash bindings

### PR DESCRIPTION
**Pull request by lmctv**
The PR implements both the one-shot API and a hashlib like one completely fulfilling #178 

I experienced test failures stemming from missing python2.6 and python3.3 in newly created environments; since I've split out in #200 the installation of those missing interpreters, failures in testing can be expected.

### CODE REVIEW

**Comment by lmctv** [comment]
@warner would you mind taking a look to this branch? If at all possible, I'd rather have this upstreamed and then I would add the final needed touches to #206

**Comment by lmctv** [comment]
@warner , @reaperhulk would you mind trying to review at last this PR and #199 ? Thank you, and sorry for being this insistent.

File: src/nacl/hashlib.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+_b2b_init = nacl.bindings.crypto_generichash_blake2b_init
+_b2b_final = nacl.bindings.crypto_generichash_blake2b_final
+_b2b_copy = nacl.bindings.crypto_generichash_blake2b_state_copy
+_b2b_update = nacl.bindings.crypto_generichash_blake2b_update
+
+
+class blake2b(object):
+    """
+    :py:mod:`hashlib` API compatible blake2b algorithm implementation
```
Comment (reaperhulk) [inline src/nacl/hashlib.py]: I'd prefer to add a hashlib compatible layer in a separate PR to make this easier to review

File: src/nacl/bindings/crypto_generichash.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+def generichash_blake2b_salt_personal(data,
+                                      digest_size=crypto_generichash_BYTES,
+                                      key=b'', salt=b'', personal=b''):
+    """One time hash interface"""
+    digest = ffi.new("unsigned char[]", digest_size)
+    _salt = ffi.new("char []", crypto_generichash_SALTBYTES)
+    _personal = ffi.new("char []", crypto_generichash_PERSONALBYTES)
+
+    ffi.memmove(_salt, salt, min(len(salt), crypto_generichash_SALTBYTES))
```
Comment (reaperhulk) [inline src/nacl/bindings/crypto_generichash.py]: Same as the siphash PR please :)

**Comment by lmctv** [comment]
Superseded by #215 #216 #217

**Pull request closed without merging**